engine/dlib/src/dlib/buffer.cpp

```diff
@@ -616,4 +644,117 @@ namespace dmBuffer
+        Buffer* buffer = GetBuffer(g_BufferContext, hbuffer);
+        if (!buffer) {
+            return RESULT_BUFFER_INVALID;
+        }
+        if (count == 0) {
+            return RESULT_METADATA_INVALID;
+        }
+
+        Buffer::MetaData* item = FindMetaDataItem(buffer, name_hash);
+        uint32_t values_block_size = GetSizeForValueType(type)*count; // do this once
+        if (item) {
+            // make sure the type and value count is right
+            if (item->m_ValueCount != count || item->m_ValueType != type) {
+                return RESULT_METADATA_INVALID; // TODO use a proper error status hsere
+            }
+            memcpy(item->m_Data, data, values_block_size);
+        } else {
+            dmArray<Buffer::MetaData*>& metadata_items = buffer->m_MetaDataArray;
+            if (metadata_items.Remaining() <= 0) {
+                metadata_items.OffsetCapacity(2); // TODO - shall we set an upper limit to the number of meta data entries ?
+            }
+            if (metadata_items.Remaining() > 0) {
+                item = (Buffer::MetaData*) malloc(sizeof(Buffer::MetaData));
+                item->m_Name = name_hash;
+                item->m_ValueCount = count; // no restriction in max number of values
+                item->m_ValueType = type; // no validation, all ValueTypes do
+                item->m_Data = malloc(values_block_size);
+                memcpy(item->m_Data, data, values_block_size);
+                metadata_items.Push(item);
+            } else {
+                return RESULT_ALLOCATION_ERROR;
+            }
+        }
+
+        return RESULT_OK;
+    }
+
+    Result SetBounds(HBuffer hbuffer, const dmVMath::Point3& min, const dmVMath::Point3& max) {
+        Buffer* buffer = GetBuffer(g_BufferContext, hbuffer);
+        if (!buffer) {
+            return RESULT_BUFFER_INVALID;
+        }
+
+        Result r = SetMetaData(hbuffer, dmHashString64("min_aabb"), &min, 3, VALUE_TYPE_FLOAT32);
+        if (r != RESULT_OK) {
+            return r;
+        }
+        r = SetMetaData(hbuffer, dmHashString64("max_aabb"), &max, 3, VALUE_TYPE_FLOAT32);
+        if (r != RESULT_OK) {
+            return r;
+        }
+
+        return RESULT_OK;
+    }
+
+    Result GetBounds(HBuffer hbuffer, dmVMath::Point3& min, dmVMath::Point3& max) {
+        Buffer* buffer = GetBuffer(g_BufferContext, hbuffer);
+        if (!buffer) {
+            return RESULT_BUFFER_INVALID;
+        }
+
+        float* floats_p;
+        uint32_t value_count;
+        ValueType value_type;
+
+        Result r = GetMetaData(hbuffer, dmHashString64("min_aabb"), (void**) &floats_p, &value_count, &value_type);
+        if (r != RESULT_OK) {
+            return r;
+        }
+        assert(value_count == 3 && "metadata 'min_aabb' should have 3 values");
+        assert(value_type == VALUE_TYPE_FLOAT32 && "metadata 'min_aabb' should be floats");
```

Comment:
Here, since this is part of a loose api, I suggest returning the `RESULT_METADATA_INVALID` instead.

Reply:
No need to fix. GetBounds() wil be removed.